paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentConfirmationRegistry.kt

```diff
@@ -0,0 +1,15 @@
+package com.stripe.android.paymentsheet
+
+import androidx.lifecycle.SavedStateHandle
+
+internal class PaymentConfirmationRegistry(
+    private val confirmationDefinitions: List<PaymentConfirmationDefinition<*, *, *, *>>
+) {
+    fun createConfirmationMediators(
+        savedStateHandle: SavedStateHandle
+    ): List<PaymentConfirmationMediator<*, *, *, *>> {
+        return confirmationDefinitions.map { definition ->
+            PaymentConfirmationMediator(savedStateHandle, definition)
+        }
+    }
+}
```

Comment:
I was initially thinking of adding different list for pre-confirmation definitions, but I think that limits definitions to 0-1 pre-confirmation definition and 1 end-confirmation definition.  I wonder if we should update the `PaymentConfirmationDefinition` interface to indicate if it is the end state of confirmation then provide this as a list of definitions.

Reply:
We would have to ensure though that the list is ordered correctly.

Reply:
Example:
```
val registry = PaymentConfirmationRegistry(
    GooglePayConfirmationDefinition, // Not an end state definition
    BacsConfirmationDefinition, // Not an end state definition
    SepaMandateConfirmationDefinition, // Not an end state definition
    CvcRecollectionConfirmationDefinition, // Not an end state definition
    LinkConfirmationDefinition, // Not an end state definition
    ExternalPaymentMethodsConfirmationDefinition, // An end state definition
    IntentConfirmationDefinition, // An end state definition
)
```

Reply:
Plus side is we can have 1-to-many confirmation definitions